contracts/core/SedaCoreV1.sol

```diff
@@ -80,22 +115,101 @@ contract SedaCoreV1 is ISedaCore, RequestHandlerBase, ResultHandlerBase, UUPSUpg
         SedaDataTypes.Result calldata result,
         uint64 batchHeight,
         bytes32[] calldata proof
-    ) public override(ResultHandlerBase, IResultHandler) returns (bytes32) {
-        uint256 requestTimestamp = _storageV1().requestTimestamps[result.drId];
-        // Validate result timestamp comes after request timestamp
+    ) public payable override(ResultHandlerBase, IResultHandler) returns (bytes32) {
+        // Get request details from storage
+        RequestDetails memory requestDetails = _storageV1().requestDetails[result.drId];
+
+        // Check: Validate result timestamp comes after request timestamp
         // Note: requestTimestamp = 0 for requests not tracked by this contract (always passes validation)
-        if (result.blockTimestamp <= requestTimestamp) {
-            revert InvalidResultTimestamp(result.drId, result.blockTimestamp, requestTimestamp);
+        if (result.blockTimestamp <= requestDetails.timestamp) {
+            revert InvalidResultTimestamp(result.drId, result.blockTimestamp, requestDetails.timestamp);
         }
 
-        bytes32 resultId = super.postResult(result, batchHeight, proof);
+        // Call parent postResult but return the batch sender for fee distribution
+        (bytes32 resultId, address batchSender) = super.postResultAndGetBatchSender(result, batchHeight, proof);
 
+        // Clean up state
         _removeRequest(result.drId);
-        delete _storageV1().requestTimestamps[result.drId];
+        delete _storageV1().requestDetails[result.drId];
+
+        // Handle fee distribution
+        uint256 refundAmount;
+
+        if (requestDetails.requestFee > 0) {
+            address payableAddress = result.paybackAddress.length == 20
+                ? address(bytes20(result.paybackAddress))
+                : address(0);
+
+            if (payableAddress == address(0)) {
+                // If no payback address, send all request fee to requestor
+                refundAmount += requestDetails.requestFee;
+            } else {
+                // Calculate request fee split and send to payback address
+                uint256 submitterFee = (result.gasUsed * requestDetails.requestFee) / requestDetails.gasLimit;
+                if (submitterFee > 0) {
+                    _transferFee(payableAddress, submitterFee);
+                    emit FeeDistributed(result.drId, payableAddress, submitterFee, ISedaCore.FeeType.REQUEST);
+                }
+                // Update requestor amount with refund (executed at the end of the function)
```

Comment:
For example, here we could instead say:
```
//  We need to calculate the refund any remaining fees to pay back to the requestor, however,
// we can't pay out the amount till correctness is ensured.
```

It's more verbose, but imo it's significantly more clear to even a new person reading the code.